src/extension_traits/vec.rs

```diff
@@ -34,36 +37,36 @@ use ::core::slice;
 /// ```
 pub
 trait VecReserveUninit {
+    type Item;
+
     fn reserve_uninit (self: &'_ mut Self, additional: usize)
-        -> &'_ mut [MaybeUninit<u8>]
+        -> &'_ mut [MaybeUninit<Self::Item>]
     ;
 }
-impl VecReserveUninit for Vec<u8> {
+impl<T> VecReserveUninit for Vec<T> {
+    type Item = T;
+
     fn reserve_uninit (self: &'_ mut Self, additional: usize)
-        -> &'_ mut [MaybeUninit<u8>]
+        -> &'_ mut [MaybeUninit<T>]
     {
         self.reserve(additional);
         unsafe {
             // # Safety
             //
-            //  1. Vec<T> contract guarantees that after a call to `.reserve(n)`
+            //  - Vec<T> contract guarantees that after a call to `.reserve(n)`
             //     at least `n` uninitialized elements after the end of the
             //     Vec's current length can be soundly written to.
-            //
-            //  2. `Vec` also guarantees that `self.len()` does not overflow
-            //     `isize`, so neither does `self.len() * 1`
-            slice::from_raw_parts_mut( // 1.
-                utils::ptr_cast_mut::<u8, MaybeUninit<u8>>(
+            slice::from_raw_parts_mut(
+                utils::ptr_cast_mut::<T, MaybeUninit<T>>(
                     self.as_mut_ptr()
-                        .add(self.len()) // 2.
+                        .wrapping_add(self.len())
```

Comment:
Now that `T` can be bigger than a `u8`, `mem::size_of::<T>() * self.len()` _could_ overflow `isize`, right?
Hence the conservative `.wrapping_add()`